syditest/test_api.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import requests
+import re
+import sys
+import atexit
+
+# These are standard python unit tests, but are generally intended
+# to be run with trial. Trial doesn't capture logging nicely if you
+# use python 'logging': it only works if you use Twisted's own.
```

Comment:
ugh god do we have to? Can we just use a test runner that doesn't require you to change everything about the way you write code?

Reply:
Honestly I am entirely neutral on what test runner we use, as long as:
 * Its output has one test per line with what is being tested and a pass/fail (ideally in colour when appropriate)
 * It can output results to other formats (eg. TAP)

@hawkowl suggested trial which does fit the bill, other than that it doesn't capture logging output unless you use twisted's logger.

Reply:
well, let's not use TAP since xunit is superior in almost all ways, but otherwise, I guess I can live with this: it's coming down to "what is your favourite test framework", which is quite subjective but things like this make trial very much not mine.

(For the record I think you should check out `nosetests` which checks your boxes and doesn't make you buy into its own world of logging)

Reply:
ah, nose2 does with verbose mode enabled. Also 'green' does similar but with nice colours.

Reply:
ah, but they don't appear to do anything sensible with the logging: green has a -l option but I can't get it to do anything, nose2 has a plugin so might work,  but would have to faffa bout writing config for it. (trial puts logs in _trial_temp which is nice). Could investigate further - depends how strong people's opinions are.

(nosetests looks to be EOL)

Reply:
everything is awful. I hadn't realised nosetests was EOL.